components/profile-controller/controllers/profile_controller.go

```diff
@@ -325,109 +329,84 @@ func (r *ProfileReconciler) appendErrorConditionAndReturn(ctx context.Context, i
 func (r *ProfileReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&profilev1.Profile{}).
-		Owns(&istiorbac.ServiceRole{}).
-		Owns(&istiorbac.ServiceRoleBinding{}).
 		Owns(&corev1.Namespace{}).
 		Owns(&corev1.ServiceAccount{}).
 		Owns(&rbacv1.RoleBinding{}).
 		Complete(r)
 }
 
+func (r *ProfileReconciler) getSecurityCondition(profileIns *profilev1.Profile) []*istioSecurity.Condition {
+	if r.UserIdHeader == "" {
+		return []*istioSecurity.Condition{}
+	}
+	return []*istioSecurity.Condition{
+		{
+			Key: fmt.Sprintf("request.headers[%v]", r.UserIdHeader),
+			Values: []string{
+				r.UserIdPrefix + profileIns.Spec.Owner.Name,
+				"",
+			},
+		},
+	}
+}
+
+func (r *ProfileReconciler) getAuthorizationPolicy(profileIns *profilev1.Profile) istioSecurity.AuthorizationPolicy {
+	return istioSecurity.AuthorizationPolicy{
+		Rules: []*istioSecurity.Rule{
```

Comment:
to fully reflect the previous `ServiceRoleBinding` rules we should allow intra-namespace traffic as well as implemented here: https://github.com/kubeflow/kubeflow/pull/5353. Something like that:
```
From: []*istioapi.Rule_From{
  {
	  Source: &istioapi.Source{
		  Namespaces: []string{profileIns.Name},
	  },
  },
},
```

Reply:
I don't believe so. I will include it in my review.